src/pal/src/misc/perftrace.cpp

```diff
@@ -659,26 +659,29 @@ PERFlushAllLogs( )
 
 static
 void
-PERFLogFileName(PathCharString * destFileString, const char *fileName, const char *suffix, int max_length)
+PERFLogFileName(PathCharString& destFileString, const char *fileName, const char *suffix)
 {
     const char *dir_path;
     CPalThread* pThread = InternalGetCurrentThread();
-    char * destFileName = (char*)InternalMalloc(pThread, max_length);
     dir_path = (profile_log_path == NULL) ? "." : profile_log_path;
-
+        
+    destFileString.Append(dir_path, strlen(dir_path));
+    destFileString.Append(PATH_SEPARATOR, strlen(PATH_SEPARATOR));
     if (fileName != NULL)
     {
-        snprintf(destFileName, max_length, "%s%s%s", dir_path, PATH_SEPARATOR, fileName);
+        destFileString.Append(fileName, strlen(fileName));
     }
     else
     {
-        snprintf(destFileName, max_length, "%s%s%d_%d%s", dir_path, PATH_SEPARATOR,
-            program_info.process_id, THREADSilentGetCurrentThreadId(), suffix);
+        std::string process_id     = std::to_string(program_info.process_id);
```

Comment:
In this case the Append is not likely to reallocate as I allocate a little extra memory on Resize https://github.com/dotnet/coreclr/pull/3441/files#diff-cf55086fe2a13f919bda759421e145ffR41